libindi/drivers/telescope/lx200driver.cpp

```diff
@@ -875,10 +875,11 @@ int setCalenderDate(int fd, int dd, int mm, int yy)
     if ((error_type = tty_write_string(fd, read_buffer, &nbytes_write)) != TTY_OK)
         return error_type;
 
-    error_type = tty_nread_section(fd, read_buffer, RB_MAX_LEN, '#', LX200_TIMEOUT, &nbytes_read);
+ //   error_type = tty_read_section(fd, read_buffer, '#', LX200_TIMEOUT, &nbytes_read);//zy2018
     // Read the next section whih has 24 blanks and then a #
     // Can't just use the tcflush to clear the stream because it doesn't seem to work correctly on sockets
-    tty_nread_section(fd, dummy_buffer, RB_MAX_LEN, '#', LX200_TIMEOUT, &nbytes_read);
+//    tty_read_section(fd, dummy_buffer, '#', LX200_TIMEOUT, &nbytes_read);//zy2018
+	 error_type = tty_read(fd, read_buffer, 1, 3, &nbytes_read);//zy2018
```

Comment:
Please explain the reason for this.

Reply:
because this command return "1", not "#":

 Command: “:SCYYMMDD#”
 Response: “1”
 Sets the current Local Date.